src/routes/bounce_activities_page.jsx

```diff
@@ -35,15 +36,23 @@ export default class BounceActivityPage extends React.Component {
     this.handlePrevClicked = this.handlePrevClicked.bind(this);
     this.handleNextClicked = this.handleNextClicked.bind(this);
     this.handleOptionSelector = this.handleOptionSelector.bind(this);
+    this.handleClearSearch = this.handleClearSearch.bind(this);
+    this.onDateChange = this.onDateChange.bind(this);
+    this.onFocusChange = this.onFocusChange.bind(this);
+    this.filterActivities = this.filterActivities.bind(this);
   }
 
   async componentDidMount() {
+    const { currentPageIndex } = this.state;
     try {
-      const { data: activities } = await getActivityLog();
+      const { data: activities } = await getActivityLog({
+        limit: 99999,
```

Comment:
Let's assign limit to a const variable that's more readable like MAX_ACTIVITY_LOGS so we don't have magic numbers